docs/02-app/01-building-your-application/02-data-fetching/01-fetching-caching-and-revalidating.mdx

```diff
@@ -0,0 +1,315 @@
+---
+title: Data Fetching, Caching, and Revalidating
+nav_title: Fetching, Caching, and Revalidating
+description: Learn how to fetch, cache, and revalidate data in your Next.js application.
+---
+
+Data fetching is a core part of any application. This page goes through how you can fetch, cache, and revalidate data in React and Next.js.
+
+There are three main ways you can fetch data:
+
+1. On the server, with the `fetch` API.
+2. On the server, with third-party libraries.
+3. On the client, with third-party libraries.
+
+## Fetching Data on the Server with `fetch`
+
+Next.js extends the native [`fetch` Web API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) to allow you to configure the [caching](#caching-data) and [revalidating](#revalidating-data) behavior for each fetch request on the server. React extends `fetch` to automatically [memoize](/docs/app/building-your-application/data-fetching/patterns#fetching-data-where-its-needed) fetch requests while rendering a React Component tree.
+
+You can use `fetch` with [`async`/`await` in Server Components](https://github.com/acdlite/rfcs/blob/first-class-promises/text/0000-first-class-support-for-promises), in [Route Handlers](), and in [Server Actions]().
+
+For example:
+
+```tsx filename="app/page.tsx" switcher
+async function getData() {
+  const res = await fetch('https://api.example.com/...')
+  // The return value is *not* serialized
+  // You can return Date, Map, Set, etc.
+
+  if (!res.ok) {
+    // This will activate the closest `error.js` Error Boundary
+    throw new Error('Failed to fetch data')
+  }
+
+  return res.json()
+}
+
+export default async function Page() {
+  const data = await getData()
+
+  return <main></main>
+}
+```
+
+```jsx filename="app/page.js" switcher
+async function getData() {
+  const res = await fetch('https://api.example.com/...')
+  // The return value is *not* serialized
+  // You can return Date, Map, Set, etc.
+
+  if (!res.ok) {
+    // This will activate the closest `error.js` Error Boundary
+    throw new Error('Failed to fetch data')
+  }
+
+  return res.json()
+}
+
+export default async function Page() {
+  const data = await getData()
+
+  return <main></main>
+}
+```
+
+> **Good to know**:
+>
+> To use `async`/`await` in a Server Component with TypeScript, you'll need to use TypeScript `5.1.3` or higher and `@types/react` `18.2.8` or higher.
+> Next.js provides helpful functions you may need when fetching data in Server Components such as [`cookies`](/docs/app/api-reference/functions/cookies) and [`headers`](/docs/app/api-reference/functions/headers). These will cause the route to be dynamically rendered as they rely on request time information.
+
+### Caching Data
+
+Caching is the process of storing data on the server (e.g. an [Edge Network](https://vercel.com/docs/concepts/edge-network/overview)) so it doesn't need to be re-fetched from your data source each time `fetch` is called.
```

Comment:
```suggestion
Caching stores data so it doesn't need to be re-fetched from your data source each time `fetch` is called.
```

Thinking about removing this because it might make it seem like you need an edge network for caching.